### PR TITLE
Patched containerd version to address CVE-2023-25173 and CVE-2023-25153 | Upgrade k3s

### PR DESCRIPTION
**Pull request by ag6221254**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Check if there is any CVE which needs to be addressed
- [x] OPTIONAL - Follow the PR standards from the example - https://github.com/microsoft/CBL-Mariner/pull/5231
- [x] Ready to merge

---

###### Summary 

Patched containerd version to address CVE-2023-25173 and CVE-2023-25153 | Upgrade k3s

###### Change Log  

- Added the latest version of containerd which is patch version to the CVE raised. The version has been changed to 1.6.18
- Using which new vendor tar has been created and uploaded to the BLOB storage for the RPM build

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

https://github.com/microsoft/CBL-Mariner/pull/5219
###### Links to CVEs  
[CVE-2023-25173](https://nvd.nist.gov/vuln/detail/CVE-2023-25173)

[CVE-2023-25153](https://nvd.nist.gov/vuln/detail/CVE-2023-25153)
###### Test Methodology

- Tested Locally
- Pipeline build id: 
1.25.5 RPM Build Pipeline run - https://dev.azure.com/mariner-org/mariner/_build/results?buildId=342130&view=results
1.26.2 RPM build pipeline run -  https://dev.azure.com/mariner-org/mariner/_build/results?buildId=342129&view=results

### CODE REVIEW

**Review by mfrw — CHANGES_REQUESTED** [review]
I think the upgrade PR is checked in.
- https://github.com/microsoft/CBL-Mariner/pull/5219

**Comment by ag6221254** [comment]
> nk the upgrade PR is checked i

Yeah last night got to know about the CVE for containerd which needs to be corrected in the same.
Can we please take some more time to run rpm one more time and will post the results here to checkin again ?

**Comment by mfrw** [comment]
> > nk the upgrade PR is checked i
> 
> Yeah last night got to know about the CVE for containerd which needs to be corrected in the same. Can we please take some more time to run rpm one more time and will post the results here to checkin again ?

Lets reflect the same in this case in the PR Title and also fill the CVE section of the Checklist

**Comment by ag6221254** [comment]
> > > nk the upgrade PR is checked i
> > 
> > 
> > Yeah last night got to know about the CVE for containerd which needs to be corrected in the same. Can we please take some more time to run rpm one more time and will post the results here to checkin again ?
> 
> Lets reflect the same in this case in the PR Title and also fill the CVE section of the Checklist

@mfrw - I have added the checklist and also pasted the link of the RPM generation for both the versions 
Can you please review and merge this PR ?

File: SPECS/k3s/k3s-1.25.5.spec [hunk 1]
```diff
@@ -26,7 +26,7 @@ Source0:        https://github.com/k3s-io/%{name}/archive/refs/tags/v%{version}+
 # 3. cd %%{name}-%%{version}-k3s2
 # 4. go mod vendor
 # 5. pushd vendor
-# 6. git clone --single-branch --branch="v1.5.18" --depth=1 https://github.com/k3s-io/containerd
```
Comment (mfrw) [inline SPECS/k3s/k3s-1.25.5.spec:29]: Lets bump the release tag and add changelog as well.

File: SPECS/k3s/k3s.spec [hunk 2]
```diff
@@ -26,7 +26,7 @@ Source0:        https://github.com/k3s-io/%{name}/archive/refs/tags/v%{version}+
 # 3. cd %%{name}-%%{version}-k3s1
 # 4. go mod vendor
 # 5. pushd vendor
-# 6. git clone --single-branch --branch="v1.5.18" --depth=1 https://github.com/k3s-io/containerd
```
Comment (mfrw) [inline SPECS/k3s/k3s.spec:29]: Lets bump release and add changelog

File: SPECS/k3s/k3s-1.25.5.spec (hunk 1 above)
Comment (ag6221254) [reply SPECS/k3s/k3s-1.25.5.spec:29]: Done

File: SPECS/k3s/k3s.spec (hunk 2 above)
Comment (ag6221254) [reply SPECS/k3s/k3s.spec:29]: Done

**Comment by xordux** [comment]
After bumping the release version, buddy build pipelines should be run again.

**Comment by ag6221254** [comment]
> After bumping the release version, buddy build pipelines should be run again.

Yes @xordux  - The above mentioned pipelines are the new runs.

File: SPECS/k3s/k3s-1.25.5.spec [hunk 3]
```diff
@@ -82,8 +82,9 @@ exit 0
 %{install_sh}
 
 %changelog
-* Wed Apr 05 2023 Anuj Garg <anujgarg@microsoft.com> - 1.25.5-1
+* Wed Apr 05 2023 Anuj Garg <anujgarg@microsoft.com> - 1.25.5-2
```
Comment (mfrw) [inline SPECS/k3s/k3s-1.25.5.spec:85]: Please add a new entry.
We do not modify the old entries.

File: SPECS/k3s/k3s.spec [hunk 4]
```diff
@@ -82,9 +82,10 @@ exit 0
 %{install_sh}
 
 %changelog
-* Wed Apr 05 2023 Anuj Garg <anujgarg@microsoft.com> - 1.26.2-1
+* Wed Apr 05 2023 Anuj Garg <anujgarg@microsoft.com> - 1.26.2-2
```
Comment (mfrw) [inline SPECS/k3s/k3s.spec:85]: Please add a new entry.
We do not modify an old entry.

File: SPECS/k3s/k3s-1.25.5.spec (hunk 3 above)
Comment (ag6221254) [reply SPECS/k3s/k3s-1.25.5.spec]: Done

File: SPECS/k3s/k3s.spec (hunk 4 above)
Comment (ag6221254) [reply SPECS/k3s/k3s.spec]: Done

**Comment by ag6221254** [comment]
@mfrw - Thank you for the approval
I don't have permission to merge it seems 
Can you also please merge the PR ?

**Pull request merged**